### DISCUSSION TITLE
What commands to use w. Lidl/Livarno HG08131B&HG08131C Zigbee RGBCW bulbs?

### DISCUSSION THREAD

**Discussion by rhk** [arendst/Tasmota#16449] [Q&A] [answered]
I run zbbridge-tasmota 9.2.0 on Sonoff's Zigbee gateway
Program Version | 9.2.0(zbbridge)
2020-12-21T15:06:32
2_7_4_9/2.2.2-dev(38a443e)

I have been able to connect various Sonoff zb sensors (PIR, temperature), Ikea Tradfri switches as well as a Lidl/SilverCrest plug. They all work fine.

Today I bought Lidl RGBCW bulbs, and am having issues finding correct command set to control the bulbs.
Models:
E14: HG03131B
E27: HG03131C

I'm able to connect them to the bridge. I see the devices in the Tasmota Web UI list and their 'uptime'. I also was able to set the Friendly names. So then trying to have better control over the color&  brightness...

![kuva](https://user-images.githubusercontent.com/404543/188637395-e2c76b2c-20b0-474b-b593-d0de25498e3e.png)

Using the type ID I found this page describing some MQTT commands: https://www.zigbee2mqtt.io/devices/HG08131C.html
However State or Brightness commands don't work as described on the site:
12:45:36 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"State":"Off"}}
12:45:36 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: State

12:46:17 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Brightness":"250"}}
12:46:17 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: Brightness

While trying out the lights, both of them turned red very soon after the first brightness tests and I haven't figured out a way to control the colors so far, they stay red.

This leads me to believe, that zigbee2mqtt may not be the one to follow, but check the Tasmota commands:

12:48:12 CMD: ZbInfo ZbLightE14_1
12:48:12 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbInfo":{"ZbLightE14_1":{"Device":"0x8822","Name":"ZbLightE14_1","IEEEAddr":"0x0C4314FFFE7AF2C8","ModelId":"TS0505B","Manufacturer":"_TZ3000_th6zqqy6","Endpoints":[1],"Config":[],"Reachable":true,"LastSeen":134,"LastSeenEpoch":1662461158,"LinkQuality":92}}}
12:48:12 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbInfo":"Done"}

12:48:29 CMD: ZbInfo ZbLightE27_1
12:48:29 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbInfo":{"ZbLightE27_1":{"Device":"0x8762","Name":"ZbLightE27_1","IEEEAddr":"0x50325FFFFEA451F4","ModelId":"TS0505B","Manufacturer":"_TZ3000_qd7hej8u","Endpoints":[1],"Config":[],"Reachable":true,"LastSeen":337,"LastSeenEpoch":1662460972,"LinkQuality":92}}}
12:48:29 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbInfo":"Done"}

This TS0505B leads me to believe this is actually a Tuya light.. https://www.zigbee2mqtt.io/devices/TS0505B.html and https://zigbee.blakadder.com/Tuya_TS0505B.html - Tasmota supported?

**So far I've been able to turn the bulb with command 'Power' **
12:42:46 CMD: ZbSend {"Device":"ZbLightE27_1","Send":{"Power":"Off"}}
12:42:46 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:42:46 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8762","Name":"ZbLightE27_1","Command":"0006!00","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":92}}
12:42:52 CMD: ZbSend {"Device":"ZbLightE27_1","Send":{"Power":"On"}}
12:42:52 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:42:52 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8762","Name":"ZbLightE27_1","Command":"0006!01","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":92}}

**As well as change the brightness with the 'Dimmer' command **
12:43:31 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Dimmer":"250"}}
12:43:31 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:43:31 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8822","Name":"ZbLightE14_1","Command":"0008!04","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":92}}
12:43:37 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Dimmer":"20"}}
12:43:37 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:43:37 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8822","Name":"ZbLightE14_1","Command":"0008!04","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":92}}

But continuing with Tasmota Light commands https://tasmota.github.io/docs/Commands/#light doesn't bring much luck:

**Dimmer works - DimmerX doesn't**
12:54:17 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Dimmer1":"250"}}
12:54:17 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: Dimmer1
12:54:20 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Dimmer":"250"}}
12:54:20 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:54:20 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8822","Name":"ZbLightE14_1","Command":"0008!04","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":94}}

**Trying Color commands - not much success**
12:56:13 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Color1":"#2299FF"}}
12:56:13 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: Color1
12:56:18 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"Color":"#2299FF"}}
12:56:18 MQT: sonoffs/stat/zbbridge/RESULT = {"ZbSend":"Done"}
12:56:18 MQT: sonoffs/tele/zbbridge/RESULT = {"ZbResponse":{"Device":"0x8822","Name":"ZbLightE14_1","Command":"0300!07","Status":0,"StatusMessage":"SUCCESS","Endpoint":1,"LinkQuality":92}}

**Hsbcolor doesn't work**
12:58:12 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"HsbColor1":"50"}}
12:58:12 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: HsbColor1
12:58:26 CMD: ZbSend {"Device":"ZbLightE14_1","Send":{"HsbColor":"50,60,20"}}
12:58:26 MQT: sonoffs/stat/zbbridge/RESULT = Unrecognized zigbee command: HsbColor

So I'd be happy to know how to use the other ~16M colours other than red. 
Is there some conversion on some level in Tasmota that turns the commands to red channel only? 
How to figure out what commands are available, if the ones listed at Tasmota Commands or zigbee2mqtt don't work?
Is there something I haven't understood?

Thanks for any pointers & huge thanks for all the great work that has been put on Tasmota, I'm using it on about 20 devices here & there :)

**Comment by s-hadinger** [comment, marked as answer]
Hi, zigbee commands are not working the same as Tasmota commands, and you can't have a real 1-1 equivalent.

Turn on/off:
`ZbSend {"Device":"ZbLightE27_1","Send":{"Power":<0/1> or <"On"/"Off">}}`

Brightness:
`ZbSend {"Device":"ZbLightE27_1","Send":{"Dimmer":0..254>}}` the value is in range 0..254, 255 is an invalid value.

Change Hue (color):
`ZbSend {"Device":"ZbLightE27_1","Send":{"Hue":0..254>}}` range is 0..254 and corresponds to 0..360 degrees

Change Saturation (color):
`ZbSend {"Device":"ZbLightE27_1","Send":{"Sat":0..254>}}` range is 0..254, 0 is no color, 254 is pure color

You can mix Dimmer/Hue/Sat in the same command because they share the same cluster 0x0300, but On/off need to be sent in a different command.

Unfortunately I haven't yet implemented automatic conversion from #RRGGBB to Hue/Sat/Dimmer. It is more or less on the todo list.

Final point, you can set `ZbLight ZbLightE27_1,5` to indicate that the light has 5 channels, this will enable auto-probing at restart and allow for Alexa visibility.

**Reply by rhk** [reply]
Just as a reference for anyone who might find this; following commands seem to be supported:

Power: 1/0
Dimmer 0-100
CT 153-500
Hue 0-254
Sat 0-254

The devices reports stuff like this:
11:27:20 MQT: sonoffs/tele/zbbridge/SENSOR = {"ZbReceived":{"ZbLightE27_1":{"Device":"0x8762","Name":"ZbLightE27_1","Hue":201,"Sat":254,"X":0,"Y":0,"CT":153,"ColorMode":2,"Endpoint":1,"LinkQuality":8}}}

ColorMode seems to change between 0 and 2:
0 - RGB used
2 - CW used

X and Y don't work as commands.

Playing with Hue, jumping rapidly between 0 and 254 seems to take the lamp to mode where it changes the hue rapidly automatically :D

**Reply by s-hadinger** [reply]
Hi,

Dimmer does have a range of 0-254

For X/Y it works but it's not convenient. You need to convert X/Y to ints in range 0..65535 and use the command:
`ZbSend {"Device":"ZbLightE27_1","Send":{"Color":"<x>,<y>"}}` but I suggest you stay away from it.

ColorMode can also take the value `1` if in X/Y mode.

**Reply by quensen** [reply]
Colormode 0 is not RGB but HSB
Colormode 1 also exists (X,Y), see [CIE 1931 color space](https://en.wikipedia.org/wiki/CIE_1931_color_space)
Colormode 2 is CT

There are also these read-only values, read `zbsend {"device":"tuya_11","read":{"colormode":""}}`
```
x
y
colormode
```
